multinet/multinet.py

```diff
@@ -86,26 +87,47 @@ def create_graph(workspace, graph):
     if missing:
         return (missing, "400 Missing Required Parameters")
 
-    # Validate that all referenced tables exist
-    edges = list(db.db(workspace).collection(edge_table).find({}))
-    invalid_from = set(
-        [
-            edge["_from"].split("/")[0]
-            for edge in edges
-            if edge["_from"].split("/")[0] not in node_tables
-        ]
-    )
-    invalid_to = set(
-        [
-            edge["_to"].split("/")[0]
-            for edge in edges
-            if edge["_to"].split("/")[0] not in node_tables
-        ]
-    )
-
-    if invalid_from or invalid_to:
-        error = {"error": "undefined_tables", "detail": list(invalid_from | invalid_to)}
-        return (error, "400 Edge Table Validation Failed")
+    errors = []
+
+    loadedWorkspace = db.db(workspace)
+    existing_tables = set([x["name"] for x in loadedWorkspace.collections()])
+    edges = loadedWorkspace.collection(edge_table).all()
+
+    # Iterate through each edge and check for undefined tables
+    valid_tables = dict()
+    invalid_tables = set()
+    for edge in edges:
+        nodes = (edge["_from"].split("/"), edge["_to"].split("/"))
+
+        for (table, key) in nodes:
+            if table not in existing_tables:
+                invalid_tables.add(table)
+            elif table in valid_tables:
+                valid_tables[table].add(key)
+            else:
+                valid_tables[table] = {key}
+
+    if invalid_tables:
+        for table in invalid_tables:
+            errors.append(f"Reference to undefined table: {table}")
```

Comment:
These should be replaced with structured dicts, or Python class instances of some sort. Could you open an issue about this (or reference an existing issue if there is one)?

Reply:
Dependent on #105